naples/routes/auth.py

```diff
@@ -58,3 +61,35 @@ def get_token(auth_data: s.Auth, db=Depends(get_db)):
         raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Admin user can not get API token")
     log(log.INFO, "User [%s] logged in", user.email)
     return s.Token(access_token=create_access_token(user.id))
+
+
+@router.post("/sign-up", status_code=status.HTTP_201_CREATED, response_model=s.User)
+def sign_up(data: s.UserSignIn, db: Session = Depends(get_db)):
+    """Signs up a user"""
+
+    user = m.User.get_user_by_email(data.email, session=db)
+
+    if user:
+        log(log.ERROR, "User [%s] already exists", data.email)
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="User already exists")
+
+    new_user: m.User = m.User(
+        first_name=data.first_name,
+        last_name=data.last_name,
+        email=data.email,
+        password=data.password,
+    )
+    db.add(new_user)
+    db.commit()
+
+    log(log.INFO, "User [%s] signed up", new_user.email)
+
+    db_store = db.scalar(sa.select(m.Store).where(m.Store.user_id == new_user.id))
+
+    if not db_store:
```

Comment:
Why do we need to make this db request? how can there be a possibility that this just created user has a store?

Reply:
This scenario SHOULD NOT HAPPEN!!!! Any kind of this behaviour should be FIXED ASAP!